src/store/slices/isDraftSlice.ts

```diff
@@ -0,0 +1,25 @@
+import { createSlice, PayloadAction } from "@reduxjs/toolkit";
+
+interface IDraft {
+  isDraft: boolean;
+}
+
+const initialState: IDraft = {
+  isDraft: false,
+};
+
+const isDraftSlice = createSlice({
+  name: "isDraft",
+  initialState,
+  reducers: {
+    toggleIsDraft: (state) => {
+      return { ...state, isDraft: !state.isDraft };
+    },
+    setIsDraft: (state, { payload }: PayloadAction<IDraft>) => {
+      return { ...state, isDraft: payload.isDraft };
```

Comment:
if I'm not wrong one of the bigest benefit of @reduxjs/toolkit is that yo can make state.isDraft = payload.isDraft and that's all, please check documentation, if needed I will find example

Reply:
I'll look into it 🤔

Reply:
You're right, but you still can't reassign the original state, they recommend to just return the new value

From their docs:
`
A common mistake is to try assigning state = someValue directly. This will not work! This only points the local state variable to a different reference. That is neither mutating the existing state object/array in memory, nor returning an entirely new value, so Immer does not make any actual changes.`

```jsx
const initialState = []
const todosSlice = createSlice({
  name: 'todos',
  initialState,
  reducers: {
    brokenTodosLoadedReducer(state, action) {
      // ❌ ERROR: does not actually mutate or return anything new!
      state = action.payload
    },
    fixedTodosLoadedReducer(state, action) {
      // ✅ CORRECT: returns a new value to replace the old one
      return action.payload
    },
    correctResetTodosReducer(state, action) {
      // ✅ CORRECT: returns a new value to replace the old one
      return initialState
    },
  },
})
```